### DISCUSSION TITLE
YARP truncate result received from downstream service?

### DISCUSSION THREAD

**Discussion by dario-l** [dotnet/yarp#2461] [General] [answered]
I have downstream service which exposes swagger.json. That file is 700kb and on local machine everything works fine.

Unfortunately on production that file is truncated by YARP.
YARP do not report to the browser the end of the received content (stream).

Main difference is that on production we use HTTPS where browser sends HTTP/2 request. There is no `Content-Length` or `Transfer-Encoding` in headers which is ok according to https://github.com/dotnet/aspnetcore/issues/24175

For that request (for swagger) we use MapForwarder (direct forwarding method) but I do not know is that specific to that problem or not.

**Comment by MihaZupan** [comment]
Would you be able to create a minimal runnable example that reproduces this issue?

**Reply by MihaZupan** [reply, marked as answer]
Triage: The issue appears to be with the firewall infra in front of YARP, no indication of issues from our side at this time.

**Comment by dario-l** [comment]
@MihaZupan Will try.

**Comment by dario-l** [comment]
@MihaZupan I do not have repro yet but today I've check requesting this endpoint by Postman and this is really strange.

On Postman everything works fine but on any other browser (Edge, Chrome, Firefox) received data is truncated. 😵‍💫 
I'm sending exactly the same headers.

I there any possibility to send to more specific details by DM?

**Reply by MihaZupan** [reply]
Might be worth checking whether the same HTTP version is being used by Postman and the browser.
Packet captures may also be helpful here.

Feel free to email me at `miha.zupan@microsoft.com`
